### ISSUE TITLE
Start supporting the mysql database

### ISSUE THREAD

**Issue by VinaiRachakonda** [dolthub/dolt#3111]
In order to work with parameters like `--all-databases` with mysqldumps, dolt needs to start support the default `mysql` database present in any mysql server. The default tables it should support are

```
innodb_index_stats
innodb_table_stats
```

**Comment by VinaiRachakonda** [comment]
For mysqldumps the ` --ignore-table=mysql.innodb_index_stats --ignore-table=mysql.innodb_table_stats` can be useful

**Comment by VinaiRachakonda** [comment]
Yes mainly mysqldump

**Comment by jennifersp** [comment]
Dolt supports some of mysql db tables such as user and db  as grant tables to support user privileges. Any actions on mysql database will be referencing these grant tables. Mysqldump with --all-databases includes all tables of mysql database, and import this mysqldump results no change that can be accessed from Dolt client as it will be referenced as grant tables. Also, imports to Dolt supported tables will not have imported changes as it cannot be persisted.

**Comment by timsehn** [comment]
I don't think we should support the full mysql database, just the tables needed for us. That means we won't be able to import an `--all-databases` mysqldump but I think that's ok.
